test.sh

```diff
@@ -44,3 +52,16 @@ if [ "$test" != "" ]; then
 	echo 'ERROR 4'
 	exit 1
 fi
+
+# Do not accept bcrypt passwords as input (only plain-text passwords)
+
+test=$(
+	cat <<'EOF' | node ./index.js http-basic-auth --realm='"node-red"' --username='"test"' --password='"$2y$10$5TSZDldoJ7MxDZdtK/SG2O3cwORqLDhHabYlKX9OsM.W/Z/oLwKW6"'
+{"req":{"headers":{"authorization":"Basic dGVzdDokMnkkMTAkNVRTWkRsZG9KN014RFpkdEsvU0cyTzNjd09ScUxEaEhhYllsS1g5T3NNLlcvWi9vTHdLVzY="}}}
```

Comment:
It might be less opaque if the test input isn't hardcoded in base64 but just calls the `base64` command as a subshell or using a variable.

Reply:
Good idea. Here you are https://github.com/alexandrainst/node-red-http-basic-auth/pull/9/commits/c6842f50b6d144d81ef8f060760c10b981718c39